### PR TITLE
Update CMake error message in tutorials.

### LINKED ISSUE #4600: Prevent double negation in cmake error messages
In [this google group post](https://groups.google.com/forum/#!topic/dealii/rKOubgbnKKw), the configuration-time cmake error message of a tutorial lead to some confusion regarding the requirements for PETSc complex number support. The error message

```sh
<path_to_deal_II_installation>/examples/step-17$ cmake .
CMake Error at CMakeLists.txt:42 (MESSAGE):

  Error! The deal.II library found at
 <path_to_deal_II_installation> was not configured with

      DEAL_II_WITH_PETSC = ON
      DEAL_II_PETSC_WITH_COMPLEX = OFF

  One or all of these are OFF in your installation but are required for this
  tutorial step.
```

has a double negation of the second prerequisite (i.e not `DEAL_II_PETSC_WITH_COMPLEX = OFF` == `DEAL_II_PETSC_WITH_SCALAR = ON`). It would be good to find a way to better transmit this message in this instance.

### PR DESCRIPTION
**Pull request by jppelteret**
This removes any ambiguity in the outputted message previously caused by
double negation.

Fixes #4600

### CODE REVIEW

**Comment by tjhei** [comment]
Much better than before. One additional idea: would it make sense to also include what configuration the user has as in "you configured with DEAL_II_WITH_PETSC = ${DEAL_II_WITH_PETSC}".

**Comment by jppelteret** [comment]
Good idea! I’ll make this update this in a moment.

File: examples/step-40/CMakeLists.txt [hunk 1]
```diff
@@ -39,14 +39,20 @@ ENDIF()
...
     DEAL_II_WITH_TRILINOS = ON
     DEAL_II_WITH_P4EST = ON
-One or both of these combinations are OFF in your installation but at least one is required for this tutorial step."
+However, the deal.II library found at ${DEAL_II_PATH} was configured with these options
+    DEAL_II_WITH_PETSC = ${DEAL_II_WITH_PETSC}
+    DEAL_II_PETSC_WITH_COMPLEX = ${DEAL_II_PETSC_WITH_COMPLEX}
+    DEAL_II_WITH_P4EST = ${DEAL_II_WITH_P4EST}
+    DEAL_II_WITH_TRILINOS = ${DEAL_II_WITH_TRILINOS}
+which conflict with the requirements.
+One or both of the aforementioned combinations of prerequisites are not met by your installation, but at least one is required for this tutorial step."
```
Comment (jppelteret) [inline examples/step-40/CMakeLists.txt]: Take note that the outputted message is slightly more detailed here.

File: examples/step-55/CMakeLists.txt [hunk 2]
```diff
@@ -39,14 +39,20 @@ ENDIF()
...
     DEAL_II_WITH_TRILINOS = ON
     DEAL_II_WITH_P4EST = ON
-One or both of these combinations are OFF in your installation but at least one is required for this tutorial step."
+However, the deal.II library found at ${DEAL_II_PATH} was configured with these options
+    DEAL_II_WITH_PETSC = ${DEAL_II_WITH_PETSC}
+    DEAL_II_PETSC_WITH_COMPLEX = ${DEAL_II_PETSC_WITH_COMPLEX}
+    DEAL_II_WITH_P4EST = ${DEAL_II_WITH_P4EST}
+    DEAL_II_WITH_TRILINOS = ${DEAL_II_WITH_TRILINOS}
+which conflict with the requirements.
+One or both of the aforementioned combinations of prerequisites are not met by your installation, but at least one is required for this tutorial step."
```
Comment (jppelteret) [inline examples/step-55/CMakeLists.txt]: and here as well.

**Comment by tjhei** [comment]
This looks great! Does it show up as "ON" and "OFF" or how does cmake produce the output?

**Comment by tjhei** [comment]
if yes, please merge yourself.

**Comment by jppelteret** [comment]
I think that this pretty much does what we'd hoped it would. Pointing the tutorials with prerequisites to a minimal installation leads to the following error messages:

- Step-18
```
step-18/build$ cmake -DDEAL_II_DIR=<path>/install ..
CMake Error at CMakeLists.txt:42 (MESSAGE):


  Error! This tutorial requires a deal.II library that was configured with
  the following options:

      DEAL_II_WITH_MPI = ON
      DEAL_II_WITH_PETSC = ON
      DEAL_II_PETSC_WITH_COMPLEX = OFF

  However, the deal.II library found at
  <path>/install was configured with these options

      DEAL_II_WITH_MPI = OFF
      DEAL_II_WITH_PETSC = OFF
      DEAL_II_PETSC_WITH_COMPLEX =

  which conflict with the requirements.


-- Configuring incomplete, errors occurred!
```

- Step-22
```
step-22/build$ cmake -DDEAL_II_DIR=<path>/install ..
CMake Error at CMakeLists.txt:41 (MESSAGE):


  Error! This tutorial requires a deal.II library that was configured with
  the following options:

      DEAL_II_WITH_UMFPACK = ON

  However, the deal.II library found at
  <path>/install was configured with these options

      DEAL_II_WITH_UMFPACK = OFF

  which conflict with the requirements.


-- Configuring incomplete, errors occurred!
```

- Step-55
```
step-55/build$ cmake -DDEAL_II_DIR=<path>/install ..
CMake Error at CMakeLists.txt:41 (MESSAGE):


  Error! This tutorial requires a deal.II library that was configured with
  the following options:

      DEAL_II_WITH_PETSC = ON
      DEAL_II_PETSC_WITH_COMPLEX = OFF
      DEAL_II_WITH_P4EST = ON

  or

      DEAL_II_WITH_TRILINOS = ON
      DEAL_II_WITH_P4EST = ON

  However, the deal.II library found at
  <path>/install was configured with these options

      DEAL_II_WITH_PETSC = OFF
      DEAL_II_PETSC_WITH_COMPLEX =
      DEAL_II_WITH_P4EST = OFF
      DEAL_II_WITH_TRILINOS = OFF

  which conflict with the requirements.

  One or both of the aforementioned combinations of prerequisites are not met
  by your installation, but at least one is required for this tutorial step.


-- Configuring incomplete, errors occurred!
```

I don't think that there's anything that we can do about the "blank" entries, right? I gather that those are variables that simply don't exist because `PETSc` isn't configured.

**Comment by bangerth** [comment]
I think the blank entries are acceptable. I'd say this is already a significant improvement, so assuming that @tjhei 's approval still stands, I'll merge. Thanks for working through this!

**Comment by tjhei** [comment]
> 'd say this is already a significant improvement, so assuming that @tjhei 's approval still stands

:+1:

**Pull request merged**